### ISSUE TITLE
Add regression tests for being able to open the main types of files (SAV, BIQ), or at least to fail acceptably

### ISSUE THREAD

**Issue by QuintillusCFC** [C7-Game/Prototype#154]
Although we're still getting C#/Godot testing sorted, we do have at least one test.  So, it's time to start prioritizing the highest-priority stuff.

One of those is making sure we don't regress on being able to open the big categories of files:

 - SAV files
 - BIQ files
 - Subtypes of those, mainly:
   - BIQ files with/without custom rules
   - BIQ files with/without custom player data
   - BIQ files with/without random maps
   - SAVs based on BIQs in those "without" categories
 - The default C7 static-json file

Note that some of these aren't fully supported yet, e.g. BIQs without custom player rules.  In those cases, we should fail gracefully, rather than crash.

The exact form of these tests could vary, although I am thinking an integration test is more appropriate than a unit test in this case (we should also have more unit tests, but that's another story).  I'll create a separate issue for integrating these tests into GitHub workflows.

**Comment by pcen** [comment]
I can add once https://github.com/C7-Game/Prototype/pull/181 is merged since it gives a uniform interface to load saves from. We should also merge https://github.com/C7-Game/Prototype/pull/182 so we can run these without Visual Studio.
